### PR TITLE
Preprocessing

### PR DESCRIPTION
**Pull request by naveenram00**
Changed preprocessing code so that it was more generic and less repetitive. Added support for ml_tags and ml_sequences and changed finetune.py to train on combinations of the three tasks.

### CODE REVIEW

File: trainer/constants.py [hunk 1]
```diff
@@ -30,3 +30,15 @@
     "train": os.path.join(DATA_DIR, "rd-train.tsv"),
     "validation": os.path.join(DATA_DIR, "rd-validation.tsv")
 }
+ML_SEQ_TSV_PATH = {
+    "train": os.path.join(DATA_DIR, "ml-train.tsv"),
+    "validation": os.path.join(DATA_DIR, "ml-validation.tsv")
+}
+ML_TAGS_TSV_PATH = {
+    "train": os.path.join(DATA_DIR, "ml-tags-train.tsv"),
+    "validation": os.path.join(DATA_DIR, "ml-tags-test.tsv")
```
Comment (log0) [inline trainer/constants.py:39]: nit: consistent naming would be nice, but no need to fix, just fyi.

File: trainer/finetune.py [hunk 2]
```diff
@@ -31,16 +31,33 @@
 flags.DEFINE_string("name", "default", "name/description of model  version")
 flags.DEFINE_enum("mode", "all", ["train", "evaluate", "all"],
                   "run mode: train, evaluate, or all")
+flags.DEFINE_enum("task", "rd_recommendations", ["rd_recommendations",
```
Comment (log0) [inline trainer/finetune.py:34]: Expand the doc a little but if needed, as this is quite important to what they do?

File: trainer/finetune.py [hunk 3]
```diff
@@ -119,10 +188,10 @@ def main(_):
   if FLAGS.mode == "all" or FLAGS.mode == "evaluate":
     model.batch_size = train_batch_size * 4 # larger batch size to save memory.
     model.eval(
-        mixture_or_task_name="rd_recommendations",
+        mixture_or_task_name=FLAGS.task,
         checkpoint_steps="all"
     )
-    metrics.save_metrics("rd_recommendations", model_dir)
+    # metrics.save_metrics("rd_recommendations", model_dir)
```
Comment (log0) [inline trainer/finetune.py:194]: do you intend to comment this out to not save metrics?

File: trainer/constants.py (hunk 1 above)
Comment (naveenram00) [reply trainer/constants.py]: fixed it

File: trainer/finetune.py (hunk 2 above)
Comment (naveenram00) [reply trainer/finetune.py:34]: where do you think this should go? in the flag description string or in the readme or something? I think the readme might be a good place.

**Pull request merged**